### PR TITLE
Mark the _state field as atomic and move to proper atomics instead of llvmcall

### PR DESCRIPTION
**Pull request by gbaraldi**

### CODE REVIEW

**Comment by giordano** [comment]
https://buildkite.com/julialang/julia-master/builds/39245#019157be-da72-4fb0-9294-7835a6bdcbff/830-1686
```
Error in testset Serialization:
Error During Test at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-HL2F7YQ3XH.0/build/default-honeycrisp-HL2F7YQ3XH-0/julialang/julia-master/julia-3eb9ab3e1c/share/julia/test/testdefs.jl:24
  Got exception outside of a @test
  LoadError: ConcurrencyViolationError("setfield!: atomic field cannot be written non-atomically")
  Stacktrace:
    [1] setproperty!
      @ ./Base.jl:53 [inlined]
    [2] setproperty!
      @ ./task.jl:193 [inlined]
    [3] deserialize(s::Serializer{IOBuffer}, ::Type{Task})
      @ Serialization /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-HL2F7YQ3XH.0/build/default-honeycrisp-HL2F7YQ3XH-0/julialang/julia-master/julia-3eb9ab3e1c/share/julia/stdlib/v1.12/Serialization/src/Serialization.jl:1575
```
looks relevant

**Comment by vchuravy** [comment]
Fixes https://github.com/JuliaLang/julia/pull/55125#discussion_r1687352130

See #55125 in general for which tests to update. I think you need to simply adjust the serialization code to also use an atomic here.

**Comment by gbaraldi** [comment]
Looking at the uses of _state most are unordered atomics so I won't change them. But they might deserve a look

File: stdlib/Serialization/src/Serialization.jl [hunk 1]
```diff
@@ -1570,11 +1570,11 @@ function deserialize(s::AbstractSerializer, ::Type{Task})
     t.storage = deserialize(s)
     state = deserialize(s)
     if state === :runnable
-        t._state = Base.task_state_runnable
+       @atomic :release t._state = Base.task_state_runnable
```
Comment (vchuravy) [inline stdlib/Serialization/src/Serialization.jl:1573]:
```suggestion
        @atomic :release t._state = Base.task_state_runnable
```

**Review by vchuravy — APPROVED** [review]

**Comment by giordano** [comment]
I presume this is good to go?

**Pull request merged**